go-controller/pkg/util/go_ovn.go

```diff
@@ -0,0 +1,543 @@
+package util
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"syscall"
+
+	goovn "github.com/ebay/go-ovn"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util/ovnbindings"
+	"k8s.io/klog"
+)
+
+type OVNClient struct {
+	client goovn.Client
```

Comment:
why not just embed the struct? then you don't have to write passthroughs for all the methods?

Reply:
embedding the struct will only mean that `OVNClient` implements `goovn.Client` interface, it doesn't imply that it will implement the `ovnbindings.OVNInterface`. No?

Reply:
@abhat I tend to agree with Casey here. All the OVNClient class methods (from a quick look) are passed through except for two: LSPGetDynamicAddresses() and Execute(), which only do a tiny bit of validation before passing through. If there were other fields in the OVNClient struct I'd have a different opinion.

I don't think that small amount of checking outweighs all the rest of the passthrough code...

Reply:
@abhat how about https://github.com/dcbw/ovn-kubernetes/commit/d9f5a21b6c860c808699e7b84418b6040c47382c on top of your existing branch? I know why you had the interface stuff (so that you could fake out OvnCommand) but we can actually just stuff everything we need into Exe instead.

Reply:
In principle, yes, you can stuff everything into the `Exe` of the `goovn.OvnCommand`. Thanks for the changes 👍.
I am just not sure if we need to go that far though, given we still need to have `OVNInterface`. The way I see it, having `OVNCommandInterface` decouples ovn-k completely from the go-ovn implementation. So in future if we decide that go-ovn bindings implementation can be replaced by another go library, it will be much easier. Having that extra indirection is the cost of loose coupling.

Reply:
@abhat Is there no way that upstream go-ovn can take the interface bits and mkae goovn.OvnClient implement that interface, so that we don't have to?

Reply:
`goovn.Client` is actually an interface already, but one that ties itself to `goovn.OvnCommand`. The point I was making was to decouple our ovn-k code from the underlying go-ovn library implementation altogether. Say tomorrow there is a better performing go ovn library than the one we currently are pulling in, there will be literally no code changes needed in ovn-k beyond adding support for the initialization of the new library's clients. I am OK with getting rid of the indirection layer, but in my opinion it's better for pluggability.

Reply:
Thinking about this some more, I realized that even the OVNInterface has return values for some of the methods using goovn data types. Given that is the case, I think we are still coupled with goovn at the hip. I will pull in your changes @dcbw and make the `OVNClient` and the interface indirection go away.

Reply:
Got rid of the OVNInterface and the OVNCommandInterface altogether.